### PR TITLE
fix AsyncSeq.scanAsync #11

### PR DESCRIPTION
**Pull request by eulerfx**

### CODE REVIEW

File: src/FSharp.Control.AsyncSeq/AsyncSeq.fs [hunk 1]
```diff
@@ -306,14 +306,17 @@ module AsyncSeq =
...
+    let rec go f state s = asyncSeq {
+      let! v = s
+      match v with
+      | Nil -> ()
+      | Cons(h, t) ->
         let! v = f state h
         yield v
-        yield! t |> scanAsync f v }
+        yield! t |> go f v }
+    Cons(state, go f state input) |> async.Return
```
Comment (dsyme) [inline src/FSharp.Control.AsyncSeq/AsyncSeq.fs]: I think it's better to stay idiomatic and use 

    asyncSeq { yield state; yield! go f state input }

**Comment by dsyme** [comment]
One comment, otherwise LGTM

**Comment by eulerfx** [comment]
Done. This would make for an easier change to `IAsyncEnumerable`. Maybe a cons function makes sense:

```fsharp
  let inline cons (head:'T) (tail:AsyncSeq<'T>) : AsyncSeq<'T> =
    asyncSeq { yield head ; yield! tail }
```

**Comment by dsyme** [comment]
I think you should feel free to submit PRs for correct and tested versions of any of the Seq.* functions or async variations of them (e.g. mapAsync - though we already have that).

**Pull request merged**